R/netzschleuder.R

```diff
@@ -0,0 +1,80 @@
+#' Download a graph from the Netzschleuder data catalogue
+#' Netzschleuder (<https://networks.skewed.de/>) is a large online repository for
+#' network datasets with the aim of aiding scientific research.
+#' @param name character. name of the network dataset.
+#' @param net character. If the dataset contains several networks this is the network name.
+#' @return a named list containing an edge list and node attribute data frame and some metadata
+#' @export
+read_from_netzschleuder <- function(name, net = NULL) {
+  if (is.null(net)) {
+    net <- name
+  }
+  zip_url <- paste0(
+    "https://networks.skewed.de/net/",
+    name,
+    "/files/",
+    net,
+    ".csv.zip"
+  )
+
+  temp <- tempfile()
+  utils::download.file(zip_url, temp, quiet = TRUE) #TODO: add better error handling
+  zip_contents <- utils::unzip(temp, list = TRUE)
+
+  edge_file_name <- zip_contents$Name[grepl("edge", zip_contents$Name)]
+  node_file_name <- zip_contents$Name[grepl("node", zip_contents$Name)]
+  meta_file_name <- zip_contents$Name[grepl("gprops", zip_contents$Name)]
+
+  edges_df <- utils::read.csv(unz(temp, edge_file_name)) + 1
```

Comment:
Careful: this tries to add 1 to attributes as well.

Reply:
thats what I just realized too